Build log can ring the terminal bell

**Describe the bug**

When running a nix build, both the log from the `-L` option and the final build log tail (if there was an error) passes ASCII BEL characters through, allowing a build to cause the terminal to beep (nominally, 'ring the bell'). I believe this is undesirable.

**Steps To Reproduce**

Build this package

```nix
# beeper.nix

with import <nixpkgs> {};

runCommand "beeper" {} ''
  printf "I'm going to beep\a"
''
```

Piping the build output into `cat -v` reveals that the `\a` (i.e. BEL character) is passed through, and shown as `^G`:

```
$ nix-build beeper.nix |& cat -v
beeper> I'm going to beep^G
beeper> builder for '/nix/store/438dczzqnmzpyd6j647kshjfsgifbj8r-beeper.drv' failed to produce output path '/nix/store/nrnc9q1dh9zxz8ra0ckdg8ya2gar9fwm-beeper'
error: --- Error --- build-remote
build of '/nix/store/438dczzqnmzpyd6j647kshjfsgifbj8r-beeper.drv' on 'ssh://aluminum' failed: builder for '/nix/store/438dczzqnmzpyd6j647kshjfsgifbj8r-beeper.drv' failed to produce output path '/nix/store/nrnc9q1dh9zxz8ra0ckdg8ya2gar9fwm-beeper'
error: error: --- Error --- nix-daemon
builder for '/nix/store/438dczzqnmzpyd6j647kshjfsgifbj8r-beeper.drv' failed with exit code 1; last 2 log lines:
  I'm going to beep^G
  builder for '/nix/store/438dczzqnmzpyd6j647kshjfsgifbj8r-beeper.drv' failed to produce output path '/nix/store/nrnc9q1dh9zxz8ra0ckdg8ya2gar9fwm-beeper'
```

**Expected behavior**

The build goes through silently, with the BEL filtered out or otherwise silenced.

**`nix-env --version` output**

```
nix-env (Nix) 2.4pre20201205_a5d85d0
```

This is significant because the GCC may attempt to output a hyperlink to relevant documentation (See [GCC documentation](https://gcc.gnu.org/onlinedocs/gcc-10.1.0/gcc/Diagnostic-Message-Formatting-Options.html#index-TERM_005fURLS-environment-variable)). I'll try to come up with an example. The syntax of a terminal hyperlink is (basically) `\e]8;;{link url}\a{link text}\e]8;;\a`. Note the occurence of `\a` here. See https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda

As an example, this package:

```nix
with import <nixpkgs> {};

stdenv.mkDerivation {
  name = "beeper";

  unpackPhase = ''
    cat > foo.c <<END
    int main() {
        int a; return a;
    }
    END
  '';

  buildPhase = ''
    gcc -Wall -Wextra foo.c -o foo
  '';
}
```

Produces this as part of the build log (`^G` represents BEL here as the output was piped though `cat -v`):

```plain
foo.c:2:19: warning: 'a' is used uninitialized in this function [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wuninitialized^G-Wuninitialized8;;^G]
    2 |     int a; return a;
      |                   ^
```

--- a/src/libutil/util.cc
+++ b/src/libutil/util.cc
@@ -1412,7 +1412,7 @@ std::string filterANSIEscapes(const std::string & s, bool filterAll, unsigned in
             }
         }
 
-        else if (*i == '\r')
+        else if (*i == '\r' || *i == '\a')
             // do nothing for now
             i++;
 

